pkg/infrastructure/gcp/clusterapi/firewallrules.go

```diff
@@ -295,11 +295,17 @@ func createFirewallRules(ctx context.Context, in clusterapi.InfraReadyInput, net
 		return err
 	}
 
+	machineCIDR := in.InstallConfig.Config.Networking.MachineNetwork[0].CIDR.String()
 	// api rules are needed to access the kube-apiserver on master nodes
 	firewallName = fmt.Sprintf("%s-api", in.InfraID)
 	srcTags = []string{}
 	targetTags = []string{masterTag}
-	srcRanges = []string{}
+	if in.InstallConfig.Config.Publish == types.ExternalPublishingStrategy {
+		srcRanges = []string{}
+	} else {
+		// For Internal, limit the source to the machineCIDR
+		srcRanges = []string{machineCIDR}
+	}
```

Comment:
Need to check mixed publishing as well, and we have a function for that:

https://github.com/openshift/installer/blob/e8f8bf91b0dc2c0488485bd0777ed7f6c658f771/pkg/types/installconfig.go#L581-L593

```suggestion
	srcRanges = []string{}
	if !in.InstallConfig.Config.PublicAPI() {
             srcRanges = append(srcRanges, machineCIDR)
```

Reply:
Thanks. Updated.